hardware/ZWaveBase.cpp

```diff
@@ -338,19 +338,27 @@ void ZWaveBase::SendDevice2Domoticz(const _tZWaveDevice *pDevice)
 	ID3=(unsigned char)pDevice->nodeID&0xFF;
 	ID4=pDevice->instanceID;
 
-	char szID[10];
-	sprintf(szID,"%X%02X%02X%02X", ID1, ID2, ID3, ID4);
-
 	unsigned long lID = (ID1 << 24) + (ID2 << 16) + (ID3 << 8) + ID4;
+	char szID[10];
+	sprintf(szID, "%08x", (unsigned int)lID);
 
 	int BatLevel = 255;
+
 	if ((pDevice->hasBattery) && (pDevice->batValue != 0))
 	{
 		BatLevel = pDevice->batValue;
 	}
 
 	if ((pDevice->devType==ZDTYPE_SWITCH_NORMAL)||(pDevice->devType==ZDTYPE_SWITCH_DIMMER)||(pDevice->devType== ZDTYPE_CENTRAL_SCENE))
 	{
+		bool bInvertLevel = false;
+		std::vector<std::vector<std::string> > result;
+		result = m_sql.safe_query(
+			"SELECT ID FROM DeviceStatus WHERE (DeviceID == '%q') AND (Type == %d) AND (SubType == %d) AND (SwitchType == %d)",
```

Comment:
Looking up values in the database is a very costly operation.
This patch should be handled by the web interface (webserver.cpp, gui html/js and/or mainworker.cpp, switchlight function)